src/lib/provable/gadgets/sha256.ts

```diff
@@ -239,7 +234,7 @@ function sigma(u: UInt32, bits: TupleN<number, 3>, firstShifted = false) {
  *
  * @returns The updated intermediate hash values after compression.
  */
-function sha256Compression(H: UInt32[], W: UInt32[]) {
+function sha256Compression([...H]: UInt32[], W: UInt32[]) {
```

Comment:
this fixes a bug, the compression function mutated its input, which bit me because in dynamic sha2 you not necessarily want to use the output